pkg/openstackconfigversion/git_util.go

```diff
@@ -228,3 +228,12 @@ func SyncGit(
 
 	return configVersions, nil
 }
+
+// truncateDiff  truncate the diff size to less than 2MB
+func truncateDiff(diff string, log logr.Logger) string {
+	if len(diff) > 2097151 {
+		log.Info("Git diff was truncatated as it is larger than 2MB.")
+		return diff[:2097151]
```

Comment:
I'd also add that info to the object diff data so something like:

`return fmt.Sprintf("Git diff was truncated to 512KB.\n %s", diff[:524800])`